### ISSUE TITLE
[Question] futur development

### ISSUE THREAD

**Issue by JulioJu** [GabLeRoux/webextensions-history-browser#6]
Hi

Thanks for your work. What is your plan for the future development ? 

I believe a good feature who was in Norwell was the query search, especially the « - » operator. Example: « history -google ». It's a very good feature, not implemented in the built-in Firefox history. 

Do you think implement this soon ?

**Comment by GabLeRoux** [comment]
For future development, I plan to do what I listed in [ReadMe's todo](https://github.com/GabLeRoux/webextensions-history-browser#todo).

For improved search, I wonder if there's some query language that could be easily included to do exactly this. I did a quick search in npm registry, but I'm not sure what keyword I should use. I'm not sure yet how I could implement `-` search yet.

I could quickly include [datatables multi-filter](https://datatables.net/examples/api/multi_filter.html) which adds search fields to all columns.

My personal use case when I needed this was to have a date range picker to quickly see history based on a date range. I created the following [feature/date-range-picker](https://github.com/GabLeRoux/webextensions-history-browser/tree/feature/date-range-picker) branch which I will update with master shortly.

**Comment by JulioJu** [comment]
Yes, I've seen your ReadME, but I only ask if you plan  to continue the dev and when ? When do you think to implement Query search ? 

I've readen the Norvell source code. If I know trumpai code, he use Regex. To answer to your question posted 8 month ago at https://addons.mozilla.org/en-US/firefox/addon/norwell/reviews/, to see source code simply download the extension and unzip it. As Trumpai says, this code is already Open Source. If you want reuse portions of his code, you must ask him to place it under MIT licence. 

There is no dependencies in the Trumpai plugin. It's cool, because I think it's lightweight. Maybe it's not necessary to search in npm registry

If I understand the Norwell plugin, he push only the result who match the regex.
* For search  « history -google »:
  * retrieve all history who match « history » and who don't match « google ». 
  * i.e. `result = /^.+history.+$/ - /^+google.+$/`
* For search « "history google" » (surround by double quotes or simple quote) 
  * retrieve all history who match /^.+history google.+$/. 

For the date, if I understand the Norwell plugin, he use the javascript Date Object. Each column is associated to a Date. It displays only raw with Date => minDate && Date <= maxDate.

Maybe (it's not sure) in one month I'll could try to help you. But I'm not really sure to have any time. There is not emergency, actually we have Firefox ESR for legacy plugins ;-), or Waterfox (but I don't know if it is very secure).

**Comment by JulioJu** [comment]
For query search I think something like 
``javascript
    let result = [];
    allHistory.forEach(function(item){
        date = retrieveDate(item);
        if (date >=Date && date <=Date && isMatch(item, regex)){
            result.push[item];
        }
    })
```
I don't know if it's possible to do something like that.

**Comment by JulioJu** [comment]
I try to write a pull request ;-)

**Comment by GabLeRoux** [comment]
I like the idea. Maybe we could have a `enable regex search` checkbox so one could search using a regex string directly.

I also plan to add some unit tests to the project just to make additions keep working on the long term. Project is already set to run them automatically in travis.

**Comment by JulioJu** [comment]
I've had not a much time this afternoon. The next week, I will not have lot of time too, but probably some hour the next week-end.

I believe we must customize https://datatables.net/plug-ins/filtering/row-based/range_dates#Example .
(see example at https://github.com/DataTables/DataTables/blob/master/examples/plug-ins/range_filtering.html). 

Could wait until the next week-end?

**Comment by GabLeRoux** [comment]
Sure, we can have a look next weekend. I did find the same links and examples as you. I also found [vedmack/yadcf (Yet Another DataTables Column Filter)](https://github.com/vedmack/yadcf) which has a few filters. Its date range filter uses jquery-ui and it feels old, but it could be a first working step for the date range.

Thanks for investigating this, have an awesome week.

**Comment by vedmack** [comment]
yadcf also support bootstrap datetimepicker with moment, [read more in the docs - datepicker_type](https://github.com/vedmack/yadcf/blob/933e77c612bf33f68933cb4413b4c3650929ed51/jquery.dataTables.yadcf.js#L242) [relevant showcase page](http://yadcf-showcase.appspot.com/dom_bootstrap.html) and I hope to finish the integration of another bootstrap plugin soon https://github.com/vedmack/yadcf/issues/435
